extractor/src/main/java/org/schabi/newpipe/extractor/services/youtube/YoutubeParsingHelper.java

```diff
@@ -551,4 +548,116 @@ public static void defaultAlertsCheck(final JsonObject initialData) throws Parsi
+            if (infoPanelContentRenderer.getArray("paragraphs").size() > 1) {
+                sb.append("<br>");
+            }
+        }
+        metaInfo.setText(sb.toString());
+        if (infoPanelContentRenderer.has("sourceEndpoint")) {
+            final String metaInfoLinkUrl = YoutubeParsingHelper.getUrlFromNavigationEndpoint(
+                    infoPanelContentRenderer.getObject("sourceEndpoint"));
+            try {
+                metaInfo.addUrl(new URL(Objects.requireNonNull(extractedCachedUrlIfNeeded(metaInfoLinkUrl))));
+            } catch (final NullPointerException | MalformedURLException e) {
+                throw new ParsingException("Could not get metadata info URL", e);
+            }
+
+            final String metaInfoLinkText = YoutubeParsingHelper.getTextFromObject(
+                    infoPanelContentRenderer.getObject("inlineSource"));
+            if (isNullOrEmpty(metaInfoLinkText)) {
+                throw new ParsingException("Could not get metadata info link text.");
+            }
+            metaInfo.addUrlText(metaInfoLinkText);
+        }
+
+        return metaInfo;
+    }
+
+    @Nonnull
+    private static MetaInfo getClarificationRendererContent(final JsonObject clarificationRenderer)
+            throws ParsingException {
+        final MetaInfo metaInfo = new MetaInfo();
+
+        final String title = YoutubeParsingHelper.getTextFromObject(clarificationRenderer.getObject("contentTitle"));
+        final String text = YoutubeParsingHelper.getTextFromObject(clarificationRenderer.getObject("text"));
+        if (title == null || text ==  null) {
+            throw new ParsingException("Could not extract clarification renderer content");
+        }
+        metaInfo.setTitle(title);
+        metaInfo.setText(text);
+
+        if (clarificationRenderer.has("actionButton")) {
+            final JsonObject actionButton = clarificationRenderer.getObject("actionButton")
+                    .getObject("buttonRenderer");
+            try {
+                final String url = YoutubeParsingHelper.getUrlFromNavigationEndpoint(actionButton.getObject("command"));
+                metaInfo.addUrl(new URL(Objects.requireNonNull(extractedCachedUrlIfNeeded(url))));
+            } catch (final NullPointerException | MalformedURLException e) {
+                throw new ParsingException("Could not get metadata info URL", e);
+            }
+
+            final String metaInfoLinkText = YoutubeParsingHelper.getTextFromObject(
+                    actionButton.getObject("text"));
+            if (isNullOrEmpty(metaInfoLinkText)) {
+                throw new ParsingException("Could not get metadata info link text.");
+            }
+            metaInfo.addUrlText(metaInfoLinkText);
+        }
+
+        if (clarificationRenderer.has("secondaryEndpoint")) {
+            // TODO: implement
+            // These URLs point to a Google search of the topic
```

Comment:
This is not true, at least for me the secondary endpoint for e.g. [covid 19](https://www.youtube.com/results?search_query=covid+19) points to [Learn about vaccine progress from the WHO](https://www.who.int/emergencies/diseases/novel-coronavirus-2019/covid-19-vaccines). So I think you should add another `url` and another `urlText`. You could check with a rule of thumb if it is a google search or not (e.g. by looking for `google` inside the url), and only add it if it is not.

Reply:
I cannot reproduce that. Can you provide a JSON for me?

Reply:
I opened YouTube in a Private Window and this is what I get.
```json
{
	"clarificationRenderer": {
		"text": {
			"runs": [
				{
					"text": "Get the latest information from the WHO about coronavirus."
				}
			]
		},
		"endpoint": {
			"clickTrackingParams": "CLkBEMChAhgAIhMImZiDhenP7QIVWoV8Ch2TvAxl",
			"commandMetadata": {
				"webCommandMetadata": {
					"url": "https://www.who.int/emergencies/diseases/novel-coronavirus-2019",
					"webPageType": "WEB_PAGE_TYPE_UNKNOWN",
					"rootVe": 83769
				}
			},
			"urlEndpoint": {
				"url": "https://www.who.int/emergencies/diseases/novel-coronavirus-2019",
				"target": "TARGET_NEW_WINDOW"
			}
		},
		"menu": {
			"menuRenderer": {
				"items": [
					{
						"menuNavigationItemRenderer": {
							"text": {
								"runs": [
									{
										"text": "Why am I seeing this?"
									}
								]
							},
							"icon": {
								"iconType": "INFO"
							},
							"navigationEndpoint": {
								"clickTrackingParams": "CLkBEMChAhgAIhMImZiDhenP7QIVWoV8Ch2TvAxl",
								"commandMetadata": {
									"webCommandMetadata": {
										"url": "//support.google.com/youtube/answer/9004474?hl=en-GB",
										"webPageType": "WEB_PAGE_TYPE_UNKNOWN",
										"rootVe": 83769
									}
								},
								"urlEndpoint": {
									"url": "//support.google.com/youtube/answer/9004474?hl=en-GB",
									"target": "TARGET_NEW_WINDOW"
								}
							},
							"trackingParams": "CLkBEMChAhgAIhMImZiDhenP7QIVWoV8Ch2TvAxl",
							"accessibility": {
								"accessibilityData": {
									"label": "Why am I seeing this?"
								}
							}
						}
					},
					{
						"menuNavigationItemRenderer": {
							"text": {
								"runs": [
									{
										"text": "Send feedback"
									}
								]
							},
							"icon": {
								"iconType": "FEEDBACK"
							},
							"navigationEndpoint": {
								"clickTrackingParams": "CLkBEMChAhgAIhMImZiDhenP7QIVWoV8Ch2TvAxl",
								"commandMetadata": {
									"webCommandMetadata": {
										"ignoreNavigation": true
									}
								},
								"userFeedbackEndpoint": {
									"additionalDatas": [
										{
											"userFeedbackEndpointProductSpecificValueData": {
												"key": "clarify_id",
												"value": "/g/11j2cc_qll"
											}
										}
									],
									"bucketIdentifier": "clarify_box"
								}
							},
							"trackingParams": "CLkBEMChAhgAIhMImZiDhenP7QIVWoV8Ch2TvAxl",
							"accessibility": {
								"accessibilityData": {
									"label": "Send feedback",
									"identifier": {
										"accessibilityIdType": "MENU_FEEDBACK"
									}
								}
							}
						}
					}
				],
				"trackingParams": "CLkBEMChAhgAIhMImZiDhenP7QIVWoV8Ch2TvAxl",
				"accessibility": {
					"accessibilityData": {
						"label": "Action menu"
					}
				}
			}
		},
		"source": {
			"simpleText": "Learn about vaccine progress from the WHO"
		},
		"trackingParams": "CLkBEMChAhgAIhMImZiDhenP7QIVWoV8Ch2TvAxl",
		"style": "CLARIFICATION_RENDERER_STYLE_MEDIUM",
		"contentTitle": {
			"runs": [
				{
					"text": "COVID-19"
				}
			]
		},
		"secondaryEndpoint": {
			"clickTrackingParams": "CLkBEMChAhgAIhMImZiDhenP7QIVWoV8Ch2TvAxl",
			"commandMetadata": {
				"webCommandMetadata": {
					"url": "https://www.who.int/emergencies/diseases/novel-coronavirus-2019/covid-19-vaccines",
					"webPageType": "WEB_PAGE_TYPE_UNKNOWN",
					"rootVe": 83769
				}
			},
			"urlEndpoint": {
				"url": "https://www.who.int/emergencies/diseases/novel-coronavirus-2019/covid-19-vaccines",
				"target": "TARGET_NEW_WINDOW"
			}
		},
		"actionButton": {
			"buttonRenderer": {
				"style": "STYLE_SUGGESTIVE",
				"size": "SIZE_DEFAULT",
				"isDisabled": false,
				"text": {
					"simpleText": "LEARN MORE"
				},
				"trackingParams": "CLoBEPBbIhMImZiDhenP7QIVWoV8Ch2TvAxl",
				"command": {
					"clickTrackingParams": "CLoBEPBbIhMImZiDhenP7QIVWoV8Ch2TvAxl",
					"commandMetadata": {
						"webCommandMetadata": {
							"url": "https://www.who.int/emergencies/diseases/novel-coronavirus-2019",
							"webPageType": "WEB_PAGE_TYPE_UNKNOWN",
							"rootVe": 83769
						}
					},
					"urlEndpoint": {
						"url": "https://www.who.int/emergencies/diseases/novel-coronavirus-2019",
						"target": "TARGET_NEW_WINDOW"
					}
				}
			}
		},
		"background": "CLARIFICATION_RENDERER_BACKGROUND_INFO",
		"secondarySource": {
			"simpleText": "Learn about vaccine progress from the WHO"
		},
		"loggingDirectives": {
			"trackingParams": "CLkBEMChAhgAIhMImZiDhenP7QIVWoV8Ch2TvAxl",
			"visibility": {
				"types": "12"
			},
			"gestures": {
				"types": "4"
			}
		}
	}
}
```

Reply:
thanks. Implemented